.github/workflows/ci-core-reusable.yml

```diff
@@ -442,11 +450,27 @@ jobs:
         run: |
           ci_run ./bin/run_on_all_chains.sh "zkstack dev test integration --no-deps --ignore-prerequisites --external-node" ${{ env.CHAINS }} ${{ env.INTEGRATION_TESTS_LOGS_DIR }}
 
+      - name: Repeat integration tests en on push to main to check for flakiness
+        if: ${{ (steps.condition.outputs.should_run == 'true' && github.event_name == 'push' && github.ref == 'refs/heads/main') }}
+        run: |
+          for i in {1..10}; do
+            echo "Iteration $i"
+            ci_run ./bin/run_on_all_chains.sh "zkstack dev test integration --no-deps --ignore-prerequisites --external-node" ${{ env.CHAINS }} ${{ env.INTEGRATION_TESTS_LOGS_DIR }}
+          done
```

Comment:
We should bump EN's data retention time as it is 3600s by default which might not be enough for running tests 10 times. I got burned by this in one of my experiments ([job](https://github.com/matter-labs/zksync-era/actions/runs/12920045398/job/36031706533?pr=3520)).